core/src/main/scala/kafka/common/InterBrokerSendThread.scala

```diff
@@ -84,9 +88,102 @@ abstract class InterBrokerSendThread(name: String,
     }
   }
 
-  def wakeup(): Unit = networkClient.wakeup()
+  private def checkDisconnects(now: Long): Unit = {
+    // any disconnects affecting requests that have already been transmitted will be handled
+    // by NetworkClient, so we just need to check whether connections for any of the unsent
+    // requests have been disconnected; if they have, then we complete the corresponding future
+    // and set the disconnect flag in the ClientResponse
+    for (node <- unsentRequests.nodes.asScala) {
+      if (networkClient.connectionFailed(node)) {
+        // Remove entry before invoking request callback to avoid callbacks handling
+        // coordinator failures traversing the unsent list again.
+        val requests = unsentRequests.remove(node)
```

Comment:
I think we might need to be careful here with concurrent modification since we're iterating this collection. We probably don't actually have the same issue that the consumer has since we don't expose the client disconnect.